marathon/src/main/scala/io/buoyant/marathon/v2/Api.scala

```diff
@@ -175,7 +175,7 @@ object Api {
     }
 }
 
-private class AppIdApi(client: Api.Client, apiPrefix: String, useHealthCheck: Boolean)
+private class AppIdApi(client: Api.Client, apiPrefix: String, useHealthCheck: Boolean, httpAuth: String)
```

Comment:
I think this should be an Option[String] throughout and only add the header if the Option is a Some.